src/SurveyService.Validation/Option/CreateOptionRequestValidator.cs

```diff
@@ -0,0 +1,26 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Models.Dto;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Option;
+
+public class CreateOptionRequestValidator : AbstractValidator<CreateOptionRequest>, ICreateOptionRequestValidator
+{
+  private readonly IQuestionRepository _questionRepository;
+  public CreateOptionRequestValidator(
+    IQuestionRepository questionRepository)
+  {
+    _questionRepository = questionRepository;
+
+    RuleFor(option => option.Content)
+      .Cascade(CascadeMode.Stop)
+      .NotEmpty().WithMessage("Content can't be empty.")
+      .MinimumLength(1).WithMessage("Content is too short.")
+      .MaximumLength(300).WithMessage("Content is too long.");
+
+    RuleFor(option => option)
+      .MustAsync(async (o, _) =>
+        (await _questionRepository.GetAsync(o.QuestionId))?.HasCustomOptions == o.IsCustom)
+      .WithMessage("Wrong IsCustom value.");
```

Comment:
It as better to use more informative and concrete messages. Something like: "This question can't have custom options."

Reply:
Done